yarn/src/main/scala/org/apache/spark/deploy/yarn/AMDelegationTokenRenewer.scala

```diff
@@ -177,6 +177,7 @@ private[yarn] class AMDelegationTokenRenewer(
     })
     // Add the temp credentials back to the original ones.
     UserGroupInformation.getCurrentUser.addCredentials(tempCreds)
+    SparkHadoopUtil.get.updateCurrentUserHDFSDelegationToken()
```

Comment:
In HA mode, there are three tokens:
1. ha token
2. namenode1 token
3. namenode2 token

Spark only update ha token.
HAUtil.cloneDelegationTokenForLogicalUri will copy ha token to namenode token.